php/libraries/User.class.inc

```diff
@@ -423,16 +423,15 @@ class User extends UserPermissions implements \LORIS\StudyEntities\AccessibleRes
     /**
      * Determines if the user has a project
      *
-     * @param int $projectID The project ID
+     * @param ProjectID $projectID The project ID
      *
      * @return bool
      */
-    public function hasProject(int $projectID) : bool
+    public function hasProject(ProjectID $projectID) : bool
     {
         return in_array(
             $projectID,
-            $this->getProjectIDs(),
-            true
```

Comment:
Loose checking can return weird results.

Reply:
It's required for the tests to pass. (See reply to other comment..)